### PR TITLE
8226938: [TEST_BUG]GTK L&F: There is no Details button in FileChooser Dialog

### PR DESCRIPTION
**Pull request by kumarabhi006**
Applet test converted to main based manual test and open sourced.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8226938](https://bugs.openjdk.org/browse/JDK-8226938): [TEST_BUG]GTK L&amp;F: There is no Details button in FileChooser Dialog (**Bug** - P4)

### Reviewers
 * [Harshitha Onkar](https://openjdk.org/census#honkar) (@honkar-jdk - **Reviewer**)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21418/head:pull/21418` \
`$ git checkout pull/21418`

Update a local copy of the PR: \
`$ git checkout pull/21418` \
`$ git pull https://git.openjdk.org/jdk.git pull/21418/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21418`

View PR using the GUI difftool: \
`$ git pr show -t 21418`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21418.diff">https://git.openjdk.org/jdk/pull/21418.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21418#issuecomment-2401346340)

### CODE REVIEW

File: test/jdk/javax/swing/JFileChooser/bug4587721.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        try {
+            UIManager.setLookAndFeel(new MetalLookAndFeel());
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+
+        String instructions = """
+                                Press the Details button.
+                                If the file name's text is chopped off by height,
+                                test failed, otherwise it succedeed.""";
```
Comment (honkar-jdk) [inline test/jdk/javax/swing/JFileChooser/bug4587721.java:54]:
```suggestion
        String instructions = """
                          Click on the Details button in JFileChooser Window.
                          If the filename text is chopped off by height,
                          then Press FAIL else Press PASS.
                          """;
```

[the remark this answers is not in the document]
Comment (kumarabhi006) [reply test/jdk/javax/swing/JFileChooser/bug4587721.java]: Ok.. updated.

File: test/jdk/javax/swing/JFileChooser/bug4587721.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import javax.swing.plaf.FontUIResource;
+import javax.swing.plaf.metal.MetalLookAndFeel;
+
+public class bug4587721 {
+
+    public static void main(String[] args) throws Exception {
+        try {
+            UIManager.setLookAndFeel(new MetalLookAndFeel());
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```
Comment (honkar-jdk) [inline test/jdk/javax/swing/JFileChooser/bug4587721.java:48]: Shouldn't this be a SkippedException?

Comment (prrace) [reply test/jdk/javax/swing/JFileChooser/bug4587721.java:48]: Metal is a standard cross-platform L&F. It absolutely should NOT be a SkippedException.
So if this fails there's a problem.
The bit that I don't get is why bother to catch it only to throw again. Just get rid of the try/catch.

Comment (kumarabhi006) [reply test/jdk/javax/swing/JFileChooser/bug4587721.java]: Updated.

**Review by honkar-jdk — APPROVED** [review]

**Review by prrace — APPROVED** [review]

**Pull request closed without merging**